### PR TITLE
Fix method confusion and change onclick.js to searchMessages.js

### PR DESCRIPTION
**Pull request by Enoch-Chou**

### CODE REVIEW

File: project/src/main/webapp/emailParse.js [hunk 1]
```diff
@@ -52,44 +41,143 @@ class MLModelEmailParse {
...
                 answer.innerHTML = mlDictAnswer["answer"];
+                answer.value = mlDictAnswer["answer"];
+                initMap();
                 console.timeEnd("Using Promises Test");
             }
         });
     }
 
-    extractEmailBodiesToArray(exampleDict) {
+    displayEmailBodies(nonEmpty) {
```
Comment (eswarty) [inline project/src/main/webapp/emailParse.js]: I thought Brandon already merged all this. Can you check with him and make sure you have the most up-to-date version from master?

**Review by eswarty — CHANGES_REQUESTED** [review]

File: project/src/main/webapp/emailParse.js (hunk 1 above)
Comment (Enoch-Chou) [reply project/src/main/webapp/emailParse.js]: yeah, sorry! Lemme get rid of these files!

File: project/src/main/webapp/gmail.js [hunk 2]
```diff
@@ -15,191 +15,237 @@
...
-    /**
-      *  On load, called to load the auth2 library and API client library.
-      */
-    function handleClientLoad() {
-    gapi.load('client:auth2', initClient);        
+
+/**
+  *  Sign out the user upon button click.
+  */
+function handleSignoutClick(event) {
```
Comment (eswarty) [inline project/src/main/webapp/gmail.js]: Are the changes to put the gmail auth code in a separate class in a different PR?

File: project/src/main/webapp/gmail.js [hunk 3]
```diff
@@ -15,191 +15,237 @@
...
+
+/**
+  *  Called when the signed in status changes, to update the UI
+  *  appropriately. After a sign-in, the API is called.
+  */
+function updateSigninStatus(isSignedIn) {
+    if (isSignedIn) {
+        authorizeButton.style.display = 'none';
+        signoutButton.style.display = 'block';
+        // getMessage();
```
Comment (eswarty) [inline project/src/main/webapp/gmail.js]: Remove commented code

File: project/src/main/webapp/gmail.js (hunk 2 above)
Comment (Enoch-Chou) [reply project/src/main/webapp/gmail.js]: different pr, this one was just to quickly push up a change that was breaking the main code

File: project/src/main/webapp/gmail.js (hunk 3 above)
Comment (Enoch-Chou) [reply project/src/main/webapp/gmail.js]: done!

File: project/src/main/webapp/gmail.js (hunk 2 above)
Comment (Enoch-Chou) [reply project/src/main/webapp/gmail.js]: the gmail auth in a separate class is in this pr: https://github.com/Enoch-Chou/capstoneproject/pull/12

File: project/src/main/webapp/gmail.js [hunk 4]
```diff
@@ -15,191 +15,235 @@
...
-                            if (emailBodyValue.indexOf("format=flowed") == -1 && emailDateValue.length > 1) {
-                                list[list.length] = emailObject;
+                            else {
+                                promiseArray.push(this.getMessage(result[i].id));
                             }
-                        });
-                    }
+                        }
+                        Promise.all(promiseArray).then(() => resolve()
+                        ); //Promise.all and then this.resolve
```
Comment (eswarty) [inline project/src/main/webapp/gmail.js]: Remove comment

File: project/src/main/webapp/gmail.js [hunk 5]
```diff
@@ -15,191 +15,235 @@
...
-            'q': "When is the interview prep workshop?"
+                'userId': 'me',
+                'q': this.question
+            });
+            getPageOfMessages(initialRequest, []);
         });
-        getPageOfMessages(initialRequest, []);
-        const t1 = performance.now();
-        console.log(t1-t0);
+        console.log("The promise ended");
```
Comment (eswarty) [inline project/src/main/webapp/gmail.js]: Remove debug logs

File: project/src/main/webapp/gmail.js [hunk 6]
```diff
@@ -15,191 +15,235 @@
...
+     */
+    getMessage(messageId) {
+        const getMessagePromise = new Promise((resolve) => {
+            const messageRequest = gapi.client.gmail.users.messages.get({
+                'userId': 'me',
+                'id': messageId,
+                'format': "raw"
+            });
+            messageRequest.execute(response => {
+                //Convert from base64 encoding to text.
```
Comment (eswarty) [inline project/src/main/webapp/gmail.js]: Nit: add space

File: project/src/main/webapp/gmail.js (hunk 2 above)
Comment (eswarty) [reply project/src/main/webapp/gmail.js]: There's also changes here updating the callbacks to promises. I'll just review those unless they're meant to be in a different PR?

File: project/src/main/webapp/gmail.js [hunk 7]
```diff
@@ -15,191 +15,235 @@
...
-  });
-  request.execute(callback);
-}
-
+    get question() {
+        return this.question;
+    }
+    get totalObjects() {
+        return this.emailObjects;
+    }
```
Comment (eswarty) [inline project/src/main/webapp/gmail.js]: Are these needed? Please add a comment if they are necessary for some reason

**Review by eswarty — CHANGES_REQUESTED** [review]

File: project/src/main/webapp/gmail.js [hunk 8]
```diff
@@ -15,191 +15,235 @@
...
+  */
+function handleSignoutClick(event) {
+    gapi.auth2.getAuthInstance().signOut();
+}
+
+/**
+  *  Initializes the API client library and sets up sign-in state
+  *  listeners.
+  */
+function initClient() {
```
Comment (eswarty) [inline project/src/main/webapp/gmail.js]: Duplicate function?

File: project/src/main/webapp/gmail.js [hunk 9]
```diff
@@ -15,191 +15,235 @@
...
+        authorizeButton.onclick = handleAuthClick;
+        signoutButton.onclick = handleSignoutClick;
+    }, function(error) {
+        appendPre(JSON.stringify(error, null, 2));
+    });
+}
+
+//Get the gmail search query from the front end
+function getQuery() {
+    const query = document.getElementById("text-input").value;
```
Comment (eswarty) [inline project/src/main/webapp/gmail.js]: Combine these lines to just:
```
return document.getElementById("text-input").value;
```

File: project/src/main/webapp/gmail.js [hunk 10]
```diff
@@ -15,191 +15,235 @@
...
+        // Handle the initial sign-in state.
+        updateSigninStatus(gapi.auth2.getAuthInstance().isSignedIn.get());
+        authorizeButton.onclick = handleAuthClick;
+        signoutButton.onclick = handleSignoutClick;
+    }, function(error) {
+        appendPre(JSON.stringify(error, null, 2));
+    });
+}
+
+//Get the gmail search query from the front end
```
Comment (eswarty) [inline project/src/main/webapp/gmail.js]: Nit: Use a JSDoc comment for function comments:
```
/** Get the gmail search query from the front end. */
```

File: project/src/main/webapp/gmail.js [hunk 11]
```diff
@@ -15,191 +15,235 @@
...
+        discoveryDocs: DISCOVERY_DOCS,
+        scope: SCOPES
+    }).then(function() {
+        // Listen for sign-in state changes.
+        gapi.auth2.getAuthInstance().isSignedIn.listen(updateSigninStatus);
+
+        // Handle the initial sign-in state.
+        updateSigninStatus(gapi.auth2.getAuthInstance().isSignedIn.get());
+        authorizeButton.onclick = handleAuthClick;
+        signoutButton.onclick = handleSignoutClick;
```
Comment (eswarty) [inline project/src/main/webapp/gmail.js]: These should be set in `index.html`
```
<button id="authorize_button" onclick="handleAuthClick()" ...>Authorize</button>
```

File: project/src/main/webapp/gmail.js [hunk 12]
```diff
@@ -15,191 +15,235 @@
...
+        if (quotedPrintableIndex != -1) {
+            emailBodyValue = emailBodyValue.substring(quotedPrintableIndex + 16);
+        }
+        return emailBodyValue;
     }
 
-    /**
-      *  Sign in the user upon button click.
-      */
-    function handleAuthClick(event) {
```
Comment (eswarty) [inline project/src/main/webapp/gmail.js]: I don't think this function should have been removed

File: project/src/main/webapp/gmail.js [hunk 13]
```diff
@@ -15,191 +15,235 @@
...
+        return emailDateValue;
     }
 
     /**
-      * Append a pre element to the body containing the given message
-      * as its text node. Used to display the results of the API call.
+      * Checks if the email is a valid email
       *
-      * @param {string} message Text to be placed in pre element.
+      * Criteria 1: Isn't a Daily Insider message - breaks ML model
```
Comment (eswarty) [inline project/src/main/webapp/gmail.js]: Commented on the other PR too but can you remove this check? I think it's better to limit the number of emails sent to the ML instead of excluding certain emails

File: project/src/main/webapp/gmail.js [hunk 14]
```diff
@@ -15,191 +15,235 @@
...
-                    request = gapi.client.gmail.users.messages.list({
-                    'userId': 'me',
-                    'pageToken': nextPageToken,
-                    'q': "When is the Interview Prep Workshop?"
-                });
-                getPageOfMessages(request, result);
-                } else {
-                    for (i = 0; i < result.length; i++) {
-                        const messageRequest = gapi.client.gmail.users.messages.get({
+    listMessages(question) {
```
Comment (eswarty) [inline project/src/main/webapp/gmail.js]: I will approve this function since you have updated it in https://github.com/Enoch-Chou/capstoneproject/pull/12/files#r474320469 and it will be a pain to revert these changes.
Just make sure that #18 gets merged before #12

**Review by eswarty — CHANGES_REQUESTED** [review]

File: project/src/main/webapp/gmail.js (hunk 7 above)
Comment (Enoch-Chou) [reply project/src/main/webapp/gmail.js]: I removed them in pull request 12. I'll get rid of them here too!

File: project/src/main/webapp/gmail.js (hunk 8 above)
Comment (Enoch-Chou) [reply project/src/main/webapp/gmail.js]: yes, thanks for catching this!

File: project/src/main/webapp/gmail.js (hunk 10 above)
Comment (Enoch-Chou) [reply project/src/main/webapp/gmail.js]: Done!

File: project/src/main/webapp/gmail.js (hunk 11 above)
Comment (Enoch-Chou) [reply project/src/main/webapp/gmail.js:56]: Could you give an example of this?

File: project/src/main/webapp/gmail.js (hunk 13 above)
Comment (eswarty) [reply project/src/main/webapp/gmail.js]: Resolved offline

**Review by eswarty — APPROVED** [review]
